framework/packages/abstract-core/src/objects/voting.rs

```diff
@@ -31,7 +31,13 @@
 //! * List of items queries allowed by `query_` prefix
 //!
 //! ## Details
-//! All methods that modify proposal will return [`ProposalInfo`] to allow logging or checking current status of proposal
+//! All methods that modify proposal will return [`ProposalInfo`] to allow logging or checking current status of proposal.
+//! 
+//! Each proposal goes through the following stages:
+//! 1. Active: proposal is active and can be voted on
+//! 2. WaitingForCount: voting period is finished and awaiting counting
+//! 3. VetoPeriod (optional): voting is counted and veto period is active
+//! 4. Finished: proposal is finished and count is done
```

Comment:
And it's 
A) Finished by count
B) Finished by veto
C) Finished by cancel during voting period

Reply:
Also 
1. can be voted or canceled on

Reply:
Thanks for adding proposal lifecycle!

Reply:
what do you mean with `canceled` on?